File: backend/app/models/label.py
from datetime import datetime

from sqlalchemy import Column, DateTime, ForeignKey, Integer, String

from app.db.database import Base


class Label(Base):
    __tablename__ = "labels"

    id = Column(
        Integer,
        primary_key=True,
        index=True,
    )

    organization_id = Column(
        Integer,
        ForeignKey(
            "organizations.id",
            ondelete="CASCADE",
        ),
        nullable=False,
    )

    name = Column(
        String(50),
        nullable=False,
    )

    color = Column(
        String(7),
        nullable=False,
    )

    created_at = Column(
        DateTime,
        default=datetime.utcnow,
        nullable=False,
    )

File: frontend/src/types/label.ts
// GET /organizations/{organization_id}/labels
export interface Label {
  id: number;
  organization_id: number;
  name: string;
  color: string;
}

// POST /organizations/{organization_id}/labels
export interface CreateLabelRequest {
  name: string;
  color: string;
}

// PATCH /labels/{label_id}
export interface UpdateLabelRequest {
  name?: string;
  color?: string;
}